Packs/HuaweiFW/ModelingRules/HuaweiFW/HuaweiFW.xif

```diff
@@ -0,0 +1,198 @@
+
+|alter 
+    xdm.event.description = if(check_brief ~= "VGADDGLOBALIP",concat(extract_msg, " Service Type: ", convert_service_type), extract_log_description ),
+    xdm.source.host.hostname = extract_fw,
+    xdm.intermediate.host.device_id = extract_id,
+    xdm.intermediate.host.hostname = extract_vsys,
+    xdm.source.user.username = extract_user,
+    xdm.source.ipv4 = extract_src_ipv4,
+    xdm.source.ipv6 = extract_src_ipv6,
+    xdm.intermediate.host.ipv4_addresses = arraycreate(extract_virtual_gw_ipv4),
+    xdm.intermediate.host.ipv6_addresses = arraycreate(extract_virtual_gw_ipv6),
+    xdm.intermediate.host.ipv4_public_addresses = arraycreate(extract_global_ipv4),
+    xdm.intermediate.host.ipv6_public_addresses = arraycreate(extract_global_ipv6),
+    xdm.event.outcome = XDM_CONST.OUTCOME_SUCCESS;
+
+
+
+call header_fields
+|filter _raw_log ~= "FWLCNS/\d/ACTIVATION"
+|alter 
+    extract_access_type = rtrim(arrayindex(regextract(extract_log_description ,"AccessType=(\S+)"),0),"\,"),
+    extract_ipv4 = arrayindex(regextract(extract_log_description ,"Ip=(\d{1,3}\.\d{1,3}\.\d{1,3}\.\d{1,3})"),0),
+    extract_ipv6 = arrayindex(regextract(extract_log_description ,"Ip=((?:[a-fA-F\d]{0,4}\:){2,7}[a-fA-F\d]{0,4})"),0),
+    extract_username = rtrim(arrayindex(regextract(extract_log_description ,"UserName=(\S+)"),0),"\)")
+
+|alter 
+    xdm.source.ipv4 = extract_ipv4,
+    xdm.source.ipv6 = extract_ipv6,
+    xdm.source.user.username = extract_username,
+    xdm.event.outcome = if(parsed_fields_headers ->log_brief = "ACTIVATION", XDM_CONST.OUTCOME_SUCCESS, parsed_fields_headers ->log_brief = "ACTIVATIONFAIL", XDM_CONST.OUTCOME_FAILED),
+    xdm.event.description = concat(extract_log_description , ", Type of logins: ", extract_access_type );
+
+
+// informational logs
+/*
+PAF/5/WRITE_LCS_FAIL, PAF/5/WRITE_PAF_FAIL, PAF/6/LCS_NOT_SAME, PAF/6/PAF_NOT_SAME, SSH/4/CMD_USERCHOICE, SSH/4/CONTINUE_KEYEXCHANGE, SSH/4/OVERWRITE_FILE, SSH/4/RENAME_FILE, SSH/4/SAVE_PUBLICKEY, SSH/4/SAVEAS_PUBLICKEY, SSH/4/STELNET_SERVER, SSH/4/UPDATE_PUBLICKEY, SYSTEM/4/ECCLMCALARM, TFTP/4/OVERWRITE, VOSCPU/4/CPU_USAGE_RESUME, VOSMEM/4/MEM_IO_USAGE_RESUME, VOSMEM/4/MEM_MAIN_USAGE_RESUME, FWLCNS/1/EXPIRED, FWLCNS/1/TRIAL, FWLCNS/1/TRIALFINISH, FWLCNS/1/TRIALWILLFINISH, FWLCNS/1/WILLEXPIRE, FWLCNS/4/EMERGNO, FWLCNS/4/EMERGOVER, FWLCNS/4/EMERGSTART, FWLCNS/4/EMERGYES, FWLCNS/4/LCSEMERG, SNMPMAC/4/LOG,SNMPMAC/4/SNMP_LOG_ARPSYNC_FULL
+*/
+call header_fields
+|filter _raw_log not contains "=" and ( _raw_log !~= "PAF/5/WRITE_LCS_FAIL" and _raw_log !~= "PAF/5/WRITE_PAF_FAIL" and _raw_log !~=  "PAF/6/LCS_NOT_SAME" and _raw_log !~=  "PAF/6/PAF_NOT_SAME") or _raw_log ~= "VOSMEM/4/MEM" or _raw_log ~= "VOSCPU/4/CPU_USAGE_HIGH"
```

Comment:
Using NOT IN would have been more suited-
https://docs-cortex.paloaltonetworks.com/r/Cortex-XSIAM/Cortex-XSIAM-Documentation/Supported-operators